### PR TITLE
Add Qt Support for Travis CI

### PR DESCRIPTION
**Pull request by yannlabou**

### CODE REVIEW

**Comment by ihhub** [comment]
Sorry, I stopped AppVeyor for your changes. I've restarted it.

**Comment by ihhub** [comment]
By the way, @yannlabou , you would be interested in [this issue](https://github.com/ihhub/penguinV/issues/204) :)

File: .gitignore [hunk 1]
```diff
@@ -33,6 +33,8 @@
 
 ## Ignore Visual Studio temporary files, build results, and
 ## files generated by popular Visual Studio add-ons.
+.vscode/
```
Comment (ihhub) [inline .gitignore]: Please revert these changes as they are not related to this issue. If you think that we have to add these files into ignore list could you please make another issue for this? It's easy to track changes if they are separate :)

File: .travis.yml [hunk 2]
```diff
@@ -13,13 +13,23 @@ matrix:
...
+    - os: linux
+      env: QT_BASE=59
     - os: linux
       env: STATIC_ANALYSIS=ON
+    - os: osx
+      env: QT_BASE=latest
 
 before_install:
-   - source script/travis/before_script.sh
+  - source script/travis/before_script.sh
```
Comment (ihhub) [inline .travis.yml]: Shall we then move this into `install` too?

File: examples/qt_framework/qimage_display/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+find_package(Qt5Widgets REQUIRED)
+set(TARGET_NAME qimage_display)
+
+# Generate code from ui files => qt5_wrap_ui
+# Generate rules for building source files from the resources => qt5_add_resources
+
+set(LIB_DIR ${CMAKE_SOURCE_DIR}/src)
+set(SRCS
+    ${LIB_DIR}/blob_detection.cpp
+    ${LIB_DIR}/image_function.cpp
```
Comment (ihhub) [inline examples/qt_framework/qimage_display/CMakeLists.txt]: Please include `${LIB_DIR}/penguinv/penguinv.cpp` file as well due to latest changes.

File: examples/qt_framework/qimage_display/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    ${LIB_DIR}/image_function_helper.cpp
+    ${LIB_DIR}/math/math_base.cpp
+    ${LIB_DIR}/FileOperation/bitmap.cpp)
+
+add_executable(${TARGET_NAME} main.cpp ${SRCS})
+target_compile_features(${TARGET_NAME} PRIVATE cxx_std_11)
+target_link_libraries(${TARGET_NAME} Qt5::Widgets)
```
Comment (ihhub) [inline examples/qt_framework/qimage_display/CMakeLists.txt]: Ann an empty line at the end of the file.

File: .travis.yml [hunk 5]
```diff
@@ -13,13 +13,23 @@ matrix:
...
+
+install:
+  - if [ "$QT_BASE" = "59" ]; then sudo apt-get install -qq qt59base; source /opt/qt59/bin/qt59-env.sh; fi
```
Comment (ihhub) [inline .travis.yml]: We need to disable CUDA installation for `QT_BASE=59` job. It fails actually. Let's separate CUDA and QT for now.

File: .travis.yml (hunk 2 above)
Comment (yannlabou) [reply .travis.yml]: Sure. I have to admit that I don't really know the difference between those two. I will check the documentation.

Comment (ihhub) [reply .travis.yml]: I think I made mistake in first place (long time ago) which is not critical. `before_install` should be something like preparation step before installation. I could be wrong of course :)

**Comment by yannlabou** [comment]
The CUDA build must have failed! I will first fix #204.

**Comment by ihhub** [comment]
> The CUDA build must have failed! I will first fix #204.

Yeah, #204 issue seems easier and requires less changes.

**Comment by yannlabou** [comment]
All review comments have been implemented. LGTM. At least on Travis CI :).

**Comment by yannlabou** [comment]
Sorry for all the commits. My other branches will be fine. I rebased my master to yours.

File: .travis.yml [hunk 6]
```diff
@@ -14,9 +14,14 @@ matrix:
     - os: linux
       env: CUDA=8.0.61-1 CMAKE_OPTIONS="-DPENGUINV_BUILD_TEST=ON"
       before_install: source script/travis/install_cuda.sh
-    - os: osx
+    - os: linux
+      env: QT_BASE=59 CMAKE_OPTIONS="-DPENGUINV_BUILD_QT_EXAMPLE=ON"
+      before_install: source script/travis/install_qt5.sh
```
Comment (ihhub) [inline .travis.yml]: Did you find out what's the difference between `install` and `before_install`? :)

Comment (yannlabou) [reply .travis.yml]: I found a very good explanation [here](https://github.com/travis-ci/travis-ci.github.com/pull/333).
TL;DR
- before_install: installs system dependencies before even changing into repo dir (e.g. apt-get update etc...)
- install: installs prerequisites or dependencies necessary to run the build

Comment (ihhub) [reply .travis.yml]: Awesome.

**Comment by ihhub** [comment]
@yannlabou thanks again for your fixes!
